apps/aecore/src/aec_pinning_agent.erl

```diff
@@ -0,0 +1,167 @@
+decode_pin_payload(<<Epoch:64/integer,Height:64/integer,BlockHash/binary>>) ->
+    #{epoch => Epoch, height => Height, block_hash => BlockHash}.
+
+
+-spec get_pins(aehttpc:node_spec(), [binary()], binary()) -> {ok, list()} | {error, term()}.
+get_pins(NodeSpec, MBs, ParentHCAccountPubKey) ->
+    Txs = lists:flatmap(
+        fun(MB) ->
+            get_hc_pins(NodeSpec, MB, ParentHCAccountPubKey)
+        end, MBs),
+    {ok, Txs}.
+
+get_hc_pins(NodeSpec, MB, ParentHCAccountPubKey) ->
+    Path =  <<"/v3/micro-blocks/hash/", MB/binary, "/transactions">>,
+    {ok, Res} = get_request(Path, NodeSpec, 5000),
+    #{<<"transactions">> := Txs} = Res,
+    ExpectedRecipient = aeser_api_encoder:encode(account_pubkey, ParentHCAccountPubKey),
+    lists:foldl(
+        fun(#{<<"tx">> := Tx}, Acc) ->
+            case Tx of
+                #{<<"type">> := <<"SpendTx">>,
+                    <<"recipient_id">> := ExpectedRecipient,
+                    <<"sender_id">> := _SenderId,
+                    <<"payload">> := PinEnc} ->
+                        {ok, Pin} = aeser_api_encoder:safe_decode(bytearray, PinEnc),
+                        DecodedPin = decode_pin_payload(Pin),
+                        [DecodedPin | Acc];
+                _ ->
+                    Acc
+            end
+        end, [], Txs).
+
+-spec create_pin_tx(binary(), binary(), binary(), integer(), integer(), binary()) -> aetx:tx().
+create_pin_tx(NodeSpec, SenderEnc, ReceiverPubkey, Amount, Fee, PinPayload) ->
+    %% 1. get the next nonce for our account over at the parent chain
+    {ok, SenderPubkey} = aeser_api_encoder:safe_decode(account_pubkey, SenderEnc),
+    %% FIXME consider altenrative approaches to fetching a nonce: ex. if there
+    %% is a hanging transaction in the pool this would produce another hanging
+    %% one
+    NoncePath = <<"/v3/accounts/", SenderEnc/binary, "/next-nonce">>,
+    {ok, #{<<"next_nonce">> := Nonce}} = get_request(NoncePath, NodeSpec, 5000),
+    %% 2. Create a SpendTx containing the Pin in its payload
+    TxArgs =
+        #{sender_id    => aeser_id:create(account, SenderPubkey),
+          recipient_id => aeser_id:create(account, ReceiverPubkey),
+          amount       => Amount,
+          %% TODO: automatic fee computation
+          %% This might not be trivial as the fee depends on a bunch of
+          %% parameters, namely:
+          %% * parent chain's protocol version
+          %% * expected min_gas on both miner and protocol level
+          %% * current gas prices
+          fee          => Fee,
+          nonce        => Nonce,
+          payload      => PinPayload},
+    {ok, SpendTx} = aec_spend_tx:new(TxArgs),
+    SpendTx.
+
+post_pin_tx(SignedSpendTx, NodeSpec) ->
+    Transaction = aeser_api_encoder:encode(transaction, aetx_sign:serialize_to_binary(SignedSpendTx)),
+    Body = #{<<"tx">> => Transaction},
+    Path = <<"/v3/transactions">>,
+    post_request(Path, Body, NodeSpec, 5000).
```

Comment:
Here you should get your transaction hash in return... no need to search for it later.

Reply:
Indeed we do, it's used in the test case.